### PR TITLE
extend UMF_DEVELOPER_MODE compiler params

### PR DESCRIPTION
**Pull request by bratpiorka**
requires #165

### CODE REVIEW

File: cmake/helpers.cmake [hunk 1]
```diff
@@ -36,6 +36,11 @@ function(add_umf_target_compile_options name)
         if(UMF_DEVELOPER_MODE)
             target_compile_options(${name} PRIVATE
                 -Werror
+                -Wextra
+                -Wshadow
+                -Wno-implicit-fallthrough
```
Comment (PatKamin) [inline cmake/helpers.cmake:41]: Do we use a fallthrough switch statement somewhere? I think the code is cleaner with explicit breaks in switches.

File: cmake/helpers.cmake [hunk 2]
```diff
@@ -36,6 +36,11 @@ function(add_umf_target_compile_options name)
         if(UMF_DEVELOPER_MODE)
             target_compile_options(${name} PRIVATE
                 -Werror
+                -Wextra
+                -Wshadow
```
Comment (igchor) [inline cmake/helpers.cmake:40]: I don't think we should use `-Wshadow` (especially for C++) where it's perfectly fine to have class/struct members named the same way as e.g. function params.

Comment (bratpiorka) [reply cmake/helpers.cmake:40]: oneCCL enables this flag during compilation and if they did this and compiled UMF, they would get errors

File: cmake/helpers.cmake (hunk 1 above)
Comment (bratpiorka) [reply cmake/helpers.cmake:41]: no, but oneCCL enables this flag so I would leave it but as you suggest - we'll probably use [[fallthrough]] attribute for such cases

File: cmake/helpers.cmake (hunk 2 above)
Comment (igchor) [reply cmake/helpers.cmake:40]: Ok, then perhaps we should add some comments why are we adding those flags?

Comment (PatKamin) [reply cmake/helpers.cmake:40]: I wonder, shouldn't it be the third party repo's problem that it enforces some compiler flags globally on included repositories such as UMF? In the end, we can't control all repos and their flags.

That's the reason why we use modified versions of `add_executable` and `add_library`. By doing this, we don't enforce flags on eventual subprojects.

**Comment by bratpiorka** [comment]
all - I'll try to change cmake options at oneCCL side instead of changing our flags. If I got problems I will re-open this PR

**Pull request closed without merging**